test/e2e_kubeadm/kubelet_config_test.go

```diff
@@ -79,46 +80,54 @@ var _ = Describe("kubelet-config ConfigMap", func() {
 			framework.Failf("error reading kubernetesVersion from %s ConfigMap: %v", kubeadmConfigName, err)
 		}
 
-		// Extract the value of the UnversionedKubeletConfigMap feature gate if its present.
-		// TODO: remove this after the UnversionedKubeletConfigMap feature gate goes GA:
-		// https://github.com/kubernetes/kubeadm/issues/1582
-		UnversionedKubeletConfigMap := true
-		if _, ok := m["featureGates"]; ok {
-			if featureGates, ok := m["featureGates"].(map[interface{}]interface{}); ok {
-				if val, ok := featureGates["UnversionedKubeletConfigMap"]; ok {
-					if valBool, ok := val.(bool); ok {
-						UnversionedKubeletConfigMap = valBool
-					} else {
-						framework.Failf("unable to cast the value of feature gate UnversionedKubeletConfigMap to bool")
-					}
-				}
-			} else {
-				framework.Failf("unable to cast the featureGates field in the %s ConfigMap", kubeadmConfigName)
-			}
-		}
-
-		// Computes all the names derived from the kubernetesVersion
 		kubeletConfigConfigMapName = "kubelet-config"
 		kubeletConfigRoleName = "kubeadm:kubelet-config"
-		// TODO: remove this after the UnversionedKubeletConfigMap feature gate goes GA:
-		// https://github.com/kubernetes/kubeadm/issues/1582
-		if !UnversionedKubeletConfigMap {
-			kubeletConfigConfigMapName = fmt.Sprintf("kubelet-config-%d.%d", k8sVersion.Major(), k8sVersion.Minor())
-			kubeletConfigRoleName = fmt.Sprintf("kubeadm:kubelet-config-%d.%d", k8sVersion.Major(), k8sVersion.Minor())
-		}
 		kubeletConfigRoleBindingName = kubeletConfigRoleName
-		kubeletConfigConfigMapResource.Name = kubeletConfigConfigMapName
+
+		kubeletConfigConfigMapNameVersioned = fmt.Sprintf("kubelet-config-%d.%d", k8sVersion.Major(), k8sVersion.Minor())
+		kubeletConfigRoleNameVersioned = fmt.Sprintf("kubeadm:kubelet-config-%d.%d", k8sVersion.Major(), k8sVersion.Minor())
+		kubeletConfigRoleBindingNameVersioned = kubeletConfigRoleNameVersioned
 	})
 
 	ginkgo.It("should exist and be properly configured", func() {
-		cm := GetConfigMap(f.ClientSet, kubeSystemNamespace, kubeletConfigConfigMapName)
-
+		// TODO: switch to GetConfigMap once UnversionedKubeletConfigMap feature gate goes GA:
+		// https://github.com/kubernetes/kubeadm/issues/1582
+		cm, err := f.ClientSet.CoreV1().
+			ConfigMaps(kubeSystemNamespace).
+			Get(context.TODO(), kubeletConfigConfigMapName, metav1.GetOptions{})
+		if err != nil {
+			cm, err = f.ClientSet.CoreV1().
+				ConfigMaps(kubeSystemNamespace).
+				Get(context.TODO(), kubeletConfigConfigMapNameVersioned, metav1.GetOptions{})
```

Comment:
just remove lines like this after the feature gate is removed.

/lgtm
hadn't thought about the kubeadm n-1 kubelet skew when reviewing last PR.
/unhold